core/domain/entities/routing/handlers/Route.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace EventEspresso\core\domain\entities\routing\handlers;
+
+use DomainException;
+use EE_Dependency_Map;
+use EventEspresso\core\domain\entities\routing\specifications\RouteMatchSpecificationInterface;
+use EventEspresso\core\services\loaders\LoaderInterface;
+use EventEspresso\core\services\request\RequestInterface;
+
+/**
+ * Class Route
+ * - class for detecting and matching with incoming requests
+ * (this can be done by directly examining the incoming Request
+ * or via a Route Match Specification class for better SRP and sharing)
+ * - registers dependencies for any classes that are required from that point forwards in the request
+ * - loads additional classes for handling the request
+ *
+ * @package EventEspresso\core\services\routing
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+abstract class Route implements RouteInterface
+{
+
+    // /** will most likely need this for routing so pencilled it in for now
+    //  * @type CommandBusInterface $command_bus
+    //  */
+    // protected $command_bus;
```

Comment:
May be remove this?

Reply:
i'm like 99.99% certain I will be implementing that... so I could remove it... but will most definitely be adding it back in